purchase_lead_hour/models/stock_orderpoint.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from dateutil import relativedelta
+
+from odoo import fields, models
+
+
+class StockOrderpoint(models.Model):
+    _inherit = "stock.warehouse.orderpoint"
+
+    def _get_product_context(self):
+        # Since product.supplierinfo delay is now a float, we need to get the
+        # available qty as of now + delay
+        res = super()._get_product_context()
+        seller = fields.first(self.product_id.seller_ids)
+        if seller:
+            lead_days, __ = self.rule_ids._get_lead_days(self.product_id)
+            availability_date = seller.get_next_availability_date()
+            res.update(
+                {
+                    "to_date": availability_date
+                    + relativedelta.relativedelta(days=lead_days)
+                }
+            )
```

Comment:
I think you could break the expected behavior by forcing the date to today here (through the call to `get_next_availability_date()`) as the std code is:
```python
            'to_date': datetime.combine(self.lead_days_date, time.max)
```
This `lead_days_date` is computed by default from today, but it can be overridden by other submodules. We could of course create a glue module, but to avoid that should we instead propose a hook to Odoo to replace the call to `fields.Date.today()` here?
```python
    @api.depends('rule_ids', 'product_id.seller_ids', 'product_id.seller_ids.delay')
    def _compute_lead_days(self):
        for orderpoint in self.with_context(bypass_delay_description=True):
            if not orderpoint.product_id or not orderpoint.location_id:
                orderpoint.lead_days_date = False
                continue
            lead_days, dummy = orderpoint.rule_ids._get_lead_days(orderpoint.product_id)
            lead_days_date = fields.Date.today() + relativedelta.relativedelta(days=lead_days)
            orderpoint.lead_days_date = lead_days_date
```

I would also need this kind of hook, it could simplify another addon here: https://github.com/OCA/stock-logistics-warehouse/pull/1411 (which BTW is incompatible with your addon, it's something we should avoid).

Reply:
I already created a hook on `product.supplierinfo`, which retrieves the seller's availability date.
Is that what you're talking about?